new-integration-tests/src/test/java/oracle/weblogic/kubernetes/ItServerStartPolicy.java

```diff
@@ -0,0 +1,1237 @@
+    patchStr.append("\"op\": \"replace\",")
+        .append(" \"path\": \"/spec/adminServer/serverStartPolicy\",")
+        .append(" \"value\":  \"NEVER\"")
+        .append(" }]");
+    logger.info("AdminServer shutdown patch string: {0}", patchStr);
+
+    patch = new V1Patch(new String(patchStr));
+    boolean crdPatched = assertDoesNotThrow(() ->
+            patchDomainCustomResource(domainUid, domainNamespace, patch, "application/json-patch+json"),
+        "patchDomainCustomResource(adminShutdown) failed");
+
+    assertTrue(crdPatched, "patchDomainCustomResource(adminShutdown) failed");
+   
+    checkPodDeleted(adminServerPodName, domainUid, domainNamespace);
+    logger.info("AdminServer shutdown success");
+
+    // check managed server pods are not affected
+    logger.info("Check managed server pods are not affected");
+    for (int i = 1; i <= replicaCount; i++) {
+      checkPodReady(managedServerPrefix + i, domainUid, domainNamespace);
+    }
+    logger.info("Managed servers are still RUNNING after AdminServer is stopped");
+    patchStr = null;
+    patchStr = new StringBuffer("[{");
+    patchStr.append("\"op\": \"replace\",")
+        .append(" \"path\": \"/spec/adminServer/serverStartPolicy\",")
+        .append(" \"value\":  \"IF_NEEDED\"")
+        .append(" }]");
+    logger.info("AdminServer restart patch string: {0}", patchStr);
+
+    patch = new V1Patch(new String(patchStr));
+    crdPatched = false;
+    crdPatched = assertDoesNotThrow(() ->
+            patchDomainCustomResource(domainUid, domainNamespace, patch, "application/json-patch+json"),
+        "patchDomainCustomResource(restart) failed");
+
+    assertTrue(crdPatched, "patchDomainCustomResource(restart) failed");
+    logger.info("Wait for admin server pod {0} to be ready in namespace {1}",
+        adminServerPodName, domainNamespace);
+    checkPodReady(adminServerPodName, domainUid, domainNamespace);
+    logger.info("Check admin service {0} is created in namespace {1}",
+        adminServerPodName, domainNamespace);
+    checkServiceExists(adminServerPodName, domainNamespace);
+    logger.info("AdminServer restart success");
+  }
+
+  /**
+   * Stop a configured cluster by patching the resource defintion with 
+   *  spec/clusters/1/serverStartPolicy set to NEVER.
+   * Make sure that ONLY servers in second Cluster are stopped. 
+   * Restart the cluster by patching the resource defintion with 
+   *  spec/clusters/1/serverStartPolicy set to IF_NEEDED.
+   * Make sure that servers in second cluster are in RUNNING state. 
+   */
+  @Test
+  @DisplayName("Restart the configured cluster with serverStartPolicy")
+  public void testConfigClusterRestart() {
+
+    String configServerPodName = domainUid + "-config-cluster-server1";
+
+    // Make sure that Configured Cluster Managed server is Ready 
+    checkPodReady(configServerPodName, domainUid, domainNamespace);
+    checkServiceExists(configServerPodName, domainNamespace);
```

Comment:
throughout the file checkServiceExists has to be called before checkPodReady

Reply:
done